Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/signature/core/storage/LocalDscStorage.kt

```diff
@@ -1,26 +1,57 @@
 package de.rki.coronawarnapp.covidcertificate.signature.core.storage
 
+import android.content.Context
 import de.rki.coronawarnapp.covidcertificate.signature.core.DscData
 import de.rki.coronawarnapp.covidcertificate.signature.core.DscDataParser
+import de.rki.coronawarnapp.util.di.AppContext
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
+import org.joda.time.Instant
 import timber.log.Timber
+import java.io.File
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class LocalDscStorage @Inject constructor(
+    @AppContext context: Context,
     private val dscDataParser: DscDataParser,
 ) {
     private val mutex = Mutex()
+    private val dscDir = File(context.filesDir, "dsc_storage")
+    private val dscFile = File(dscDir, "dsclist")
 
-    suspend fun load(): DscData = mutex.withLock {
-        Timber.d("load()")
-        throw NotImplementedError()
+    suspend fun load(): DscData? = mutex.withLock {
+        Timber.v("load()")
+
+        if (dscFile.exists()) {
+            return try {
+                dscDataParser.parse(
+                    rawData = dscFile.readBytes(),
+                    updatedAt = Instant.ofEpochMilli(dscFile.lastModified())
+                )
+            } catch (exception: Exception) {
+                Timber.e(exception, "Can't load cached data")
+                null
+            }
+        } else {
+            null
+        }
     }
 
-    suspend fun save(rawData: ByteArray?): Unit = mutex.withLock {
-        Timber.d("save(rawData=${rawData?.size}")
-        throw NotImplementedError()
+    suspend fun save(rawData: ByteArray): Unit = mutex.withLock {
+        Timber.v("save (rawData.size=${rawData.size}")
+
+        if (dscDir.mkdirs()) Timber.v("Parent folder created.")
+
+        if (dscFile.exists()) {
```

Comment:
where the file is created ? I don't see `createNewFile`

Reply:
`dscFile.writeBytes(rawData)` create the file if it doesn't exist or overwrite it if already exists. 
See `LocalDscStorageTest`